README


This archive contains the distribution Task-BeLike-RJBS,
version 20131116.000:

  be more like RJBS -- use the modules he likes!

This software is copyright (c) 2013 by Ricardo Signes.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.




Changes
Revision history for Task-BeLike-RJBS

20131116.000 2013-11-16 17:09:04 America/New_York
              stop trying to install perlbrew

              require dzil 5!

20131018.001 2013-10-18 11:40:55 America/New_York
              update bugtracker and repo

              remove Moose::Autobox

20120926.000 2012-09-26 12:09:52 America/New_York
              add App::grindperl, App::perlbrew, perl 5.14.0

              add Pod::Cpandoc

20120310.000 2012-03-10 14:04:11 America/New_York
              add Unicode::Tussle, because it's cool

              add MooseX::StrictConstructor; please use it!

              switch from List::MoreUtils to List::AllUtils

20110621.000 2011-06-21 22:16:40 America/New_York
              add Module::Which, for which_pm, which I use a lot

              add App::Uni, for uni, which I also use a lot

20110516.000 2011-05-16 16:53:43 America/New_York

              require LWP::Protocol::https to make sure I can use LWP on SSL

20110324.000 2011-03-24 11:01:14 America/New_York
              improve behavior of App::Nopaste by adding its optional prereqs

20101201.000 2010-12-01 18:21:32 America/New_York
              require perl 5.12.0

              drop CPAN, CPAN::Reporter, CPAN::Uploader
              add cpanminus and cpan-outdated

              drop Perl::Critic

              bump Log::Dispatchouli to 2

              bump Moose to 1.19

              replace Exception::Class with Throwable, Throwable::X, and
              Try::Tiny

20100611.000 2010-06-11 16:11:21 America/New_York
              drop pip, Module::Install, YAML::XS
              add perl 5.10.0
              bump up some prereqs

20091203.000 2009-12-03
              taskweaver now provides tests

20091129.000 2009-11-29 12:08:39 America/New_York
              use TaskWeaver to build bundle

20090219     2009-02-19
              dzil

1.003        2009-01-16
              update my whims
              add repo

1.002        2008-04-12
              tell smokers to skip this
              drop Rubric, which is huge and rarely needed
              require 5.008 in Makefile.PL

1.001        2008-04-07
              add version number
              add changelog
              add stupid tests

undef        2008-02-05
              first release


lib/Task/BeLike/RJBS.pm
use 5.14.0;
use warnings;
package Task::BeLike::RJBS;
{
  $Task::BeLike::RJBS::VERSION = '20131116.000';
}
# ABSTRACT: be more like RJBS -- use the modules he likes!


1;

__END__

=pod

=encoding UTF-8

=head1 NAME

Task::BeLike::RJBS - be more like RJBS -- use the modules he likes!

=head1 VERSION

version 20131116.000

=head1 TASK CONTENTS

=head2 Perl for Perl's Sake

=head3 L<perl> 5.014

Version 5.014 required because: gotta have my "package NAME BLOCK"

=head3 L<App::grindperl>

=head2 Useful Command-Line Tools

=head3 L<App::Ack> 1.82

Version 1.82 required because: a working --pager

=head3 L<App::Nopaste> 0.34

Version 0.34 required because: Gist support via Gist API v3

=head3 L<App::Whiff>

App::Whiff provides C<whiff>, which replaces C<which>, because C<which> is
stupid.

=head3 L<App::Uni>

This gets us the "uni" command, which lets you run "uni snowman" to copy and
paste the character and look funny on IRC.

=head3 L<File::Rename>

This lets me rename a bunch of files by apply a C<s///> expression, or more.

=head3 L<Unicode::Tussle>

This is a whole bunch of extra utilities for poking through Unicode data.  It's
all cool stuff.

=head2 Tools for Working with the CPAN

=head3 L<perl> 5.14.0

Version 5.14.0 required because: it has package NAME BLOCK

=head3 L<App::cpanminus> 1.1002

Version 1.1002 required because: it has --auto-cleanup

=head3 L<App::cpanoutdated> 0.12

Version 0.12 required because: it won't install old dists

=head3 L<CPAN::Mini> 0.563

=head3 L<Module::Which>

I use F<which_pm> to find the version and location of installed modules, even
if two versions are installed in different parts of C<@INC>.

=head3 L<Pod::Cpandoc>

It's like C<perldoc>, but for stuff you haven't installed (yet?).

=head2 Tools for Building CPAN Distributions

=head3 L<Dist::Zilla> 5

Version 5 required because: encoding!

=head3 L<Dist::Zilla::PluginBundle::RJBS> 1.012

Version 1.012 required because: newest available

=head3 L<Module::CoreList> 2.68

Version 2.68 required because: newest available

=head3 L<Perl::Tidy> 20071205

Version 20071205 required because: supports 5.10

=head3 L<Pod::Weaver> 4

Version 4 required because: encoding!

=head2 Application Frameworks

=head3 L<App::Cmd> 0.308

Version 0.308 required because: it has App::Cmd::Setup bugfixes

App::Cmd also gets us Getopt::Long::Descriptive.

=head2 Email-Handling Libraries

=head3 L<Email::Filter> 1.02

Version 1.02 required because: I still use it, for lack of something better

=head3 L<Email::MIME> 1.905

Version 1.905 required because: merged in Creator and Modifier modules; bug fixes

=head3 L<Email::Sender>

=head3 L<Email::Sender::Transport::SQLite>

=head2 Other Libraries I Use

=head3 L<Carp::Always>

=head3 L<Config::INI> 0.011

=head3 L<DBD::SQLite>

=head3 L<Data::GUID> 0.044

Version 0.044 required because: requires a new enough Data::UUID to work around Debian

=head3 L<DateTime> 0.51

Version 0.51 required because: provides CLDR support with fewest known bugs

=head3 L<Devel::Cover>

=head3 L<Devel::NYTProf>

=head3 L<HTML::Element> 3.22

Version 3.22 required because: has proper XML escaping

=head3 L<JSON> 2.12

Version 2.12 required because: fixes unicode handling from ASCII JSON

=head3 L<List::AllUtils>

=head3 L<Log::Dispatchouli> 2.000

Version 2.000 required because: it has Log::Dispatchouli::Global

=head3 L<Moose> 1.19

Version 1.19 required because: it has assert_coerce

=head3 L<MooseX::StrictConstructor>

All constructors must be strict!

=head3 L<namespace::autoclean>

=head3 L<PPI> 1.212

Version 1.212 required because: fixes parsing of package names with leading-digit parts

=head3 L<Params::Util> 0.38

Version 0.38 required because: has fixes to _IDENTIFIER and _CLASS

=head3 L<Plack>

=head3 L<Scalar::Util> 1.18

=head3 L<Sub::Exporter> 0.980

Version 0.980 required because: INIT collector; bug fixes

=head3 L<Term::ReadLine::Gnu> 1

Version 1 required because: improves the debugger

=head3 L<Text::Markdown> 1.0.24

Version 1.0.24 required because: has trust_list_start

=head3 L<Throwable> 0.102080

Version 0.102080 required because: StackTrace::Auto factored out

=head3 L<Throwable::X>

=head3 L<Try::Tiny> 0.007

Version 0.007 required because: exception passed to C<finally>

=head2 Sanity-Check

These are just here to make sure other things work properly.

=head3 L<Mozilla::CA>

=head3 L<Crypt::SSLeay>

=head3 L<LWP::Protocol::https>

=head3 L<IO::Socket::SSL>

=head3 L<Config::GitLike>

=head1 AUTHOR

Ricardo Signes <rjbs@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2013 by Ricardo Signes.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


Makefile.PL

use strict;
use warnings;

use 5.014000;

use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "be more like RJBS -- use the modules he likes!",
  "AUTHOR" => "Ricardo Signes <rjbs\@cpan.org>",
  "BUILD_REQUIRES" => {},
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30"
  },
  "DISTNAME" => "Task-BeLike-RJBS",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Task::BeLike::RJBS",
  "PREREQ_PM" => {
    "App::Ack" => "1.82",
    "App::Cmd" => "0.308",
    "App::Nopaste" => "0.34",
    "App::Uni" => 0,
    "App::Whiff" => 0,
    "App::cpanminus" => "1.1002",
    "App::cpanoutdated" => "0.12",
    "App::grindperl" => 0,
    "CPAN::Mini" => "0.563",
    "Carp::Always" => 0,
    "Config::GitLike" => 0,
    "Config::INI" => "0.011",
    "Crypt::SSLeay" => 0,
    "DBD::SQLite" => 0,
    "Data::GUID" => "0.044",
    "DateTime" => "0.51",
    "Devel::Cover" => 0,
    "Devel::NYTProf" => 0,
    "Dist::Zilla" => 5,
    "Dist::Zilla::PluginBundle::RJBS" => "1.012",
    "Email::Filter" => "1.02",
    "Email::MIME" => "1.905",
    "Email::Sender" => 0,
    "Email::Sender::Transport::SQLite" => 0,
    "File::Rename" => 0,
    "HTML::Element" => "3.22",
    "IO::Socket::SSL" => 0,
    "JSON" => "2.12",
    "LWP::Protocol::https" => 0,
    "List::AllUtils" => 0,
    "Log::Dispatchouli" => "2.000",
    "Module::CoreList" => "2.68",
    "Module::Which" => 0,
    "Moose" => "1.19",
    "MooseX::StrictConstructor" => 0,
    "Mozilla::CA" => 0,
    "PPI" => "1.212",
    "Params::Util" => "0.38",
    "Perl::Tidy" => 20071205,
    "Plack" => 0,
    "Pod::Cpandoc" => 0,
    "Pod::Weaver" => 4,
    "Scalar::Util" => "1.18",
    "Sub::Exporter" => "0.980",
    "Term::ReadLine::Gnu" => 1,
    "Text::Markdown" => "v1.0.24",
    "Throwable" => "0.102080",
    "Throwable::X" => 0,
    "Try::Tiny" => "0.007",
    "Unicode::Tussle" => 0,
    "namespace::autoclean" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Test::More" => "0.96",
    "strict" => 0
  },
  "VERSION" => "20131116.000",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "App::Ack" => "1.82",
  "App::Cmd" => "0.308",
  "App::Nopaste" => "0.34",
  "App::Uni" => 0,
  "App::Whiff" => 0,
  "App::cpanminus" => "1.1002",
  "App::cpanoutdated" => "0.12",
  "App::grindperl" => 0,
  "CPAN::Mini" => "0.563",
  "Carp::Always" => 0,
  "Config::GitLike" => 0,
  "Config::INI" => "0.011",
  "Crypt::SSLeay" => 0,
  "DBD::SQLite" => 0,
  "Data::GUID" => "0.044",
  "DateTime" => "0.51",
  "Devel::Cover" => 0,
  "Devel::NYTProf" => 0,
  "Dist::Zilla" => 5,
  "Dist::Zilla::PluginBundle::RJBS" => "1.012",
  "Email::Filter" => "1.02",
  "Email::MIME" => "1.905",
  "Email::Sender" => 0,
  "Email::Sender::Transport::SQLite" => 0,
  "File::Rename" => 0,
  "HTML::Element" => "3.22",
  "IO::Socket::SSL" => 0,
  "JSON" => "2.12",
  "LWP::Protocol::https" => 0,
  "List::AllUtils" => 0,
  "Log::Dispatchouli" => "2.000",
  "Module::CoreList" => "2.68",
  "Module::Which" => 0,
  "Moose" => "1.19",
  "MooseX::StrictConstructor" => 0,
  "Mozilla::CA" => 0,
  "PPI" => "1.212",
  "Params::Util" => "0.38",
  "Perl::Tidy" => 20071205,
  "Plack" => 0,
  "Pod::Cpandoc" => 0,
  "Pod::Weaver" => 4,
  "Scalar::Util" => "1.18",
  "Sub::Exporter" => "0.980",
  "Term::ReadLine::Gnu" => 1,
  "Test::More" => "0.96",
  "Text::Markdown" => "v1.0.24",
  "Throwable" => "0.102080",
  "Throwable::X" => 0,
  "Try::Tiny" => "0.007",
  "Unicode::Tussle" => 0,
  "namespace::autoclean" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



